shared/logger.py

```
@@ -12,21 +12,44 @@
+    def emit(self, record):
+        """
+        Emit a record with error handling for BlockingIOError
+        """
+        try:
+            super().emit(record)
+        except BlockingIOError:
+            # Silently ignore blocking errors to prevent application crashes
+            pass
+        except Exception:
+            # Handle other exceptions gracefully
+            pass
```

Ruff (0.14.3)

`try`-`except`-`pass` detected, consider logging the exception

(S110)